### PR TITLE
[CINFRA-162] Replicated Log Supervision

### PR DESCRIPTION
**Pull request by markuspf**
### Scope & Purpose

This is a continuation of CINFRA-162 implementing the Supervision of ReplicatedLogs.

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [x] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

CINFRA-162

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -348,8 +349,7 @@ set(ARANGODB_REPLICATION2_TEST_SOURCES
   Replication2/ReplicatedState/ReplicatedStateTest.cpp
   Replication2/ReplicatedState/StateMachineTestHelper.cpp
   Replication2/ReplicatedState/StateMachineTestHelper.h
-  Replication2/ReplicatedState/SupervisionTest.cpp
-  Replication2/ReplicatedState/StateMachines/MyStateMachine.cpp
+        Replication2/ReplicatedState/StateMachines/MyStateMachine.cpp
```
Comment (maierlars) [inline tests/CMakeLists.txt:352]:
```suggestion
  Replication2/ReplicatedState/StateMachines/MyStateMachine.cpp
```

File: arangod/Replication2/ReplicatedLog/AgencyLogSpecification.cpp [hunk 2]
```diff
@@ -228,13 +238,38 @@ auto agency::to_string(LogCurrentSupervisionElection::ErrorCode ec) noexcept
...
   LOG_TOPIC("7e572", FATAL, arangodb::Logger::REPLICATION2)
       << "Invalid LogCurrentSupervisionElection::ErrorCode "
       << static_cast<std::underlying_type_t<decltype(ec)>>(ec);
   FATAL_ERROR_ABORT();
 }
 
+auto agency::toVelocyPack(LogCurrentSupervisionElection::Outcome outcome,
+                          VPackBuilder& builder) -> void {
+  VPackObjectBuilder ob(&builder);
+  builder.add("outcome", VPackValue(static_cast<int>(outcome)));
```
Comment (maierlars) [inline arangod/Replication2/ReplicatedLog/AgencyLogSpecification.cpp:253]: you added a static string for that, right?

File: tests/js/common/shell/shell-replicated-logs-cluster.js [hunk 3]
```diff
@@ -283,6 +290,6 @@ function ReplicatedLogsWriteSuite () {
 }
 
 jsunity.run(ReplicatedLogsSuite);
-jsunity.run(ReplicatedLogsWriteSuite);
+// jsunity.run(ReplicatedLogsWriteSuite);
```
Comment (maierlars) [inline tests/js/common/shell/shell-replicated-logs-cluster.js:293]: Intentional?

**Review by maierlars — APPROVED** [review]

File: tests/js/common/shell/shell-replicated-logs-cluster.js (hunk 3 above)
Comment (markuspf) [reply tests/js/common/shell/shell-replicated-logs-cluster.js]: :scream: no

**Review by apetenchea — APPROVED** [review]

**Pull request merged**